modules/@apostrophecms/modal/ui/apos/components/TheAposModals.vue

```diff
@@ -63,17 +63,21 @@ export default {
   methods: {
     async execute(componentName, props) {
       return new Promise((resolve, reject) => {
-        this.stack.push({
+        const item = {
           id: cuid(),
           componentName,
           resolve,
           props: props || {}
-        });
+        };
+
+        this.stack.push(item);
+        apos.bus.$emit('the-apos-modals-executed', item);
```

Comment:
Just call it `modal-launched`. `the-apos-modals` is a component that can be swapped out for other implementations.